test/new-e2e/tests/process/testing.go

```diff
@@ -28,9 +28,29 @@ var processDiscoveryCheckConfigStr string
 //go:embed config/system_probe.yaml
 var systemProbeConfigStr string
 
-// assertRunningChecks asserts that the given process agent checks are running on the given VM
-func assertRunningChecks(t *assert.CollectT, vm client.VM, checks []string, withSystemProbe bool, command string) {
-	status := vm.Execute(command)
+const (
+	stressProcName    = "stress"
+	stressK8sProcName = "stress-ng-cpu"
+
+	stressK8sImageTag = "short_image:stress-ng"
+)
+
+// cleanJSONOutput removes any stray log lines before or after the JSON object in the output
+func cleanJSONOutput(str string) string {
```

Comment:
Not sure what assumptions we're making about the input here, but this can get caught in an infinite loop if the input doesn't start with { and doesn't have newlines. Sample code reproducing the issue: https://go.dev/play/p/vNl5vl_1uQR

Reply:
I think the current implementation is functionally equivalent to finding the first instance of `{` and last instance of `}` and truncating anything before/after those points, so that could be an alternative way to do it

Reply:
That's a good idea, and much simpler. Thanks!

Reply:
The new implementation looks like it has an off-by-one error that strips the leading `{` and trailing `}` from the JSON input: https://go.dev/play/p/XE1E_eYSZkP

Reply:
😅 I fixed it in the go playground and forgot to transcribe the change back to the code.